BotProject/Templates/CSharp/appsettings.json

```diff
@@ -1,4 +1,11 @@
 {
+  "feature": {
```

Comment:
can we put this into "runtime" section, i mean runtime:feature:foo

Reply:
good idea

Reply:
actually on second thought, I'm not sure I like this.

There is a difference between a setting that affects Composer and should be controlled by Composer, and a something that should come out of the normal environment.  This would be the difference, in node, of having something in the settings.json file versus in the .env file.

Reply:
just to be clear, I'm not sure we want the Composer runtime behavior configuration (runtime.enableCustom, runtime.path, runtime.command) to be mingled with things about how the runtime actually works in prod (runtime.enableFeatureX)

Reply:
feature is already a group, I don't like the idea of adding another group on top of that. We can think about a more appropriate group name if necessary.

Reply:
> just to be clear, I'm not sure we want the Composer runtime behavior configuration (runtime.enableCustom, runtime.path, runtime.command) to be mingled with things about how the runtime actually works in prod (runtime.enableFeatureX)

so those settings will only affect the prod environment, do we need to make them part of the deploy procedure, I mean, users will only need to configure those settings if they want to deploy the bot, rather than local debug the bot. Is this right?

Reply:
@benbrown @luhan2017 i'm not sure i get the point, are you suggesting that we keep a top-level feature property, which is weird because there will appear at top level of settings object. which indeed is not a top-level concept like "luis", like "app", it's a configuration to a specific variation of runtime (CSharp). so here is the schema i like the settings to be

runtime {
    type: customized | built-in   // this value will decide the behavior of plugin to attach runtime or not
    version: string; 
    features: string[]; 
    path?: string; //only valid when type = customized
    command?: string; // only valid when type = customized
}

what do you guys think?
         
    
    path: ..
    command: ...
}

is this OK, or we want something more

Reply:
actually, i think we should leave the discussion of overall settings property out this, so, in here, in your own project template, this shape is OK